static/app/views/monitors/components/timeline/utils/getAggregateStatusFromMultipleBuckets.tsx

```diff
@@ -21,3 +24,20 @@ export function getAggregateStatusFromMultipleBuckets(
       CHECKIN_STATUS_PRECEDENT[0]
     );
 }
+
+/**
+ * Given multiple stats buckets [{ok: 1, ...}, {ok: 0, ...}]
+ * returns the aggregate status across all buckets
+ */
+export function getAggregateStatusFromMultipleStatsBuckets(statsArr: StatsBucket[]) {
+  return statsArr
+    .map(getAggregateStatusFromStatsBucket)
+    .reduce(
+      (aggregateStatus, currentStatus) =>
+        CHECKIN_STATUS_PRECEDENT.indexOf(currentStatus) >
+        CHECKIN_STATUS_PRECEDENT.indexOf(aggregateStatus)
+          ? currentStatus
+          : aggregateStatus,
+      CHECKIN_STATUS_PRECEDENT[0]
+    );
+}
```

Comment:
Would be nice to see better examples in the comments here, but not blocking on this.